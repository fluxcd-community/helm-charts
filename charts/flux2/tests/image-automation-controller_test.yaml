suite: test image-automation-controller deployment
templates:
  - image-automation-controller.yaml
tests:
  - it: should be empty if source-controller is not enabled
    asserts:
      - hasDocuments:
          count: 0
    set:
      imageAutomationController.create: false
  - it: should have kind Deployment for k8s >= 1.19
    capabilities:
      majorVersion: 1
      minorVersion: 19
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - isAPIVersion:
          of: apps/v1
  - it: should set additional container args
    set:
      imageAutomationController.container.additionalArgs:
        - --testlabel1=testvalue1
    asserts:
      - contains:
          path: spec.template.spec.containers[0].args
          content: --testlabel1=testvalue1
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
  - it: should have args for Multi-tenancy lockdown
    capabilities:
      majorVersion: 1
      minorVersion: 21
    set:
      multitenancy:
        enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - isAPIVersion:
          of: apps/v1
      - contains:
          path: spec.template.spec.containers[0].args
          content: --no-cross-namespace-refs=true
  - it: should set imagePullPolicy to Always
    set:
      imageAutomationController.imagePullPolicy: Always
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
  - it: should imagePullPolicy to be default value IfNotPresent
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: should use default cluster domain when null
    set:
      clusterDomain: null
    asserts:
      - contains:
          path: spec.template.spec.containers[0].args
          content: --events-addr=http://notification-controller.$(RUNTIME_NAMESPACE).svc.cluster.local.
  - it: should use custom cluster domain
    set:
      clusterDomain: custom.domain
    asserts:
      - contains:
          path: spec.template.spec.containers[0].args
          content: --events-addr=http://notification-controller.$(RUNTIME_NAMESPACE).svc.custom.domain.
  - it: should override securityContext
    set:
      imageAutomationController.podSecurityContext:
        runAsUser: 2000
      imageAutomationController.securityContext:
        runAsUser: 3000
    asserts:
      - equal:
          path: spec.template.spec.securityContext
          value:
            runAsUser: 2000
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            runAsUser: 3000
