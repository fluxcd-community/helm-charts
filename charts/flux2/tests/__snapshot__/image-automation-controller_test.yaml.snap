should match snapshot of default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: NAMESPACE
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: flux
        app.kubernetes.io/version: 0.31.3
        control-plane: controller
        helm.sh/chart: flux2-1.0.0
      name: image-automation-controller
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: image-automation-controller
      template:
        metadata:
          annotations:
            prometheus.io/port: "8080"
            prometheus.io/scrape: "true"
          labels:
            app: image-automation-controller
        spec:
          containers:
          - args:
            - --events-addr=http://notification-controller/
            - --watch-all-namespaces=true
            - --log-level=info
            - --log-encoding=json
            - --enable-leader-election
            env:
            - name: RUNTIME_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            image: ghcr.io/fluxcd/image-automation-controller:v0.23.4
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /healthz
                port: healthz
            name: manager
            ports:
            - containerPort: 8080
              name: http-prom
            - containerPort: 9440
              name: healthz
              protocol: TCP
            readinessProbe:
              httpGet:
                path: /readyz
                port: healthz
            resources:
              limits: {}
              requests:
                cpu: 100m
                memory: 64Mi
            securityContext:
              allowPrivilegeEscalation: false
              capabilities:
                drop:
                - ALL
              readOnlyRootFilesystem: true
              runAsNonRoot: true
              seccompProfile:
                type: RuntimeDefault
            volumeMounts:
            - mountPath: /tmp
              name: temp
          securityContext:
            fsGroup: 1337
          serviceAccountName: image-automation-controller
          terminationGracePeriodSeconds: 10
          volumes:
          - emptyDir: {}
            name: temp
